test/gtest/common/main.cc

```diff
@@ -74,16 +62,9 @@ int main(int argc, char **argv) {
     }
     UCS_TEST_MESSAGE << "Using random seed of " << ucs_gtest_random_seed;
     srand(ucs_gtest_random_seed);
+
+    ucs::init_config_for_valgrind(true);
```

Comment:
move this to test_config only, save UCX_xx env vars , unset, test, and restore

Reply:
done in the new PR that adds tests for unused env vars functionality - #3613